dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java

```diff
@@ -75,16 +77,12 @@ public void validate(Reporter reporter, TrackerBundle bundle, Enrollment enrollm
         strategy.isUpdateOrDelete()
             ? bundle.getPreheat().getEnrollment(enrollment.getEnrollment()).getProgram()
             : bundle.getPreheat().getProgram(enrollment.getProgram());
-    String trackedEntity =
-        bundle.getStrategy(enrollment).isDelete()
-            ? bundle
-                .getPreheat()
-                .getEnrollment(enrollment.getEnrollment())
-                .getTrackedEntity()
-                .getUid()
-            : enrollment.getTrackedEntity();
-    OrganisationUnit ownerOrgUnit =
-        getOwnerOrganisationUnit(preheat, enrollment.getTrackedEntity(), program);
+    TrackedEntity trackedEntity =
+        bundle.getStrategy(enrollment).isUpdateOrDelete()
```

Comment:
Can you move all of this into a `getTrackedEntity()`? I don't really care how you get it when reading this top level validate method. The way its now is a bit mixed with some bundle/preheat lookup in here and some in `getTrackedEntityWhenStrategyCreate`.